nimble/host/include/host/ble_gap.h

```diff
@@ -1896,6 +1896,20 @@ int ble_gap_unpair(const ble_addr_t *peer_addr);
  */
 int ble_gap_unpair_oldest_peer(void);
 
+/**
+ * Similar to `ble_gap_unpair_oldest_peer()`, except it makes sure that current
+ * peer is not deleted.
+ *
+ * @param peer_addr             Address of the current peer (not to be deleted)
+ *
+ * @return                      0 on success;
+ *                              A BLE host HCI return code if the controller
+ *                                  rejected the request;
+ *                              A BLE host core return code on unexpected
+ *                                  error.
+ */
+int ble_gap_unpair_oldest_except_curr(const ble_addr_t *curr_peer);
```

Comment:
let us call it `ble_gap_unpair_oldest_except` () and change the parameter name to `peer_addr`. 
This functionality is not really dedicated only to current device. You might want to have device which you don't want to unpair no matter what :)

Reply:
Agreed !!